identity/client/src/components/form/JSONEditor.tsx

```diff
@@ -0,0 +1,186 @@
+ */
+
+import Editor, { useMonaco } from "@monaco-editor/react";
+import { Button, FormLabel, Stack } from "@carbon/react";
+import { observer } from "mobx-react-lite";
+import {
+  ComponentProps,
+  FC,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from "react";
+import { beautify as beautifyJSON } from "src/utility/components/editor/jsonUtils.ts";
+import { options } from "src/utility/components/editor/options.ts";
+import useTranslate from "src/utility/localization";
+import { Copy, Edit } from "@carbon/react/icons";
+import Flex from "src/components/layout/Flex.tsx";
+import { spacing03, supportError } from "@carbon/elements";
+import { useNotifications } from "src/components/notifications";
+
+type EditorFirstParam = Parameters<
+  NonNullable<ComponentProps<typeof JSONEditor>["onMount"]>
+>[0];
+
+type JSONEditorProps = {
+  value: string;
+  onChange?: (value: string) => void;
+  readOnly?: boolean;
+  onValidate?: (isValid: boolean) => void;
+  onMount?: (editor: {
+    showMarkers: () => void;
+    hideMarkers: () => void;
+  }) => void;
+};
+
+const JSONEditor: FC<JSONEditorProps> = observer(
+  ({
+    value,
+    onChange,
+    readOnly = false,
+    onValidate = () => {},
+    onMount = () => {},
+  }) => {
+    const monaco = useMonaco();
+
+    useLayoutEffect(() => {
+      monaco?.languages.json.jsonDefaults.setDiagnosticsOptions({
+        schemaValidation: "error",
+        schemaRequest: "error",
+      });
+    }, [monaco]);
+
+    return (
+      <Editor
+        options={{ ...options, readOnly }}
+        language="json"
+        value={value}
+        height="32vh"
+        width="100%"
+        onChange={(value) => {
+          onChange?.(value ?? "");
+        }}
+        onMount={(editor) => {
+          editor.focus();
+
+          onMount({
+            showMarkers: () => {
+              editor.trigger("", "editor.action.marker.next", undefined);
+              editor.trigger("", "editor.action.marker.prev", undefined);
+            },
+            hideMarkers: () => {
+              editor.trigger("", "closeMarkersNavigation", undefined);
+            },
+          });
+        }}
+        onValidate={(markers) => {
+          onValidate(markers.length === 0);
+        }}
+      />
+    );
+  },
+);
+
+type JSONEditorFieldProps = {
+  label: string;
+  value: string;
+  errors?: string[] | string;
+  readOnly?: boolean;
+  onChange?: (newValue: string) => void;
+  beautify?: boolean;
+  copy?: boolean;
+  copyProps?: { notificationText?: string; onClick?: () => void };
+};
+
+const JSONEditorField: FC<JSONEditorFieldProps> = ({
+  label,
+  value,
+  errors = [],
+  readOnly = false,
+  onChange,
+  beautify = false,
+  copy = false,
+  copyProps,
+}) => {
+  const { t } = useTranslate();
+  const { enqueueNotification } = useNotifications();
+
+  const [isValid, setIsValid] = useState(true);
+  const editorRef = useRef<EditorFirstParam | null>(null);
+
+  useEffect(() => {
+    if (isValid) {
```

Comment:
Have you tested this? So if isValid changes to false nothing happens, is that correct?